tests/test_bundle_ckpt_export.py

```diff
@@ -37,67 +38,69 @@ def setUp(self):
         self.device = os.environ.get("CUDA_VISIBLE_DEVICES")
         if not self.device:
             os.environ["CUDA_VISIBLE_DEVICES"] = "0"  # default
+        self.meta_file = os.path.join(os.path.dirname(__file__), "testing_data", "metadata.json")
+        self.config_file = os.path.join(os.path.dirname(__file__), "testing_data", "inference.json")
+        self.tempdir_obj = tempfile.TemporaryDirectory()
+        tempdir = self.tempdir_obj.name
+        self.def_args = {"meta_file": "will be replaced by `meta_file` arg"}
+        self.def_args_file = os.path.join(tempdir, "def_args.yaml")
+
+        self.ckpt_file = os.path.join(tempdir, "model.pt")
+        self.ts_file = os.path.join(tempdir, "model.ts")
+
+        self.parser = ConfigParser()
+        self.parser.export_config_file(config=self.def_args, filepath=self.def_args_file)
+        self.parser.read_config(self.config_file)
+        self.net = self.parser.get_parsed_content("network_def")
 
     def tearDown(self):
         if self.device is not None:
             os.environ["CUDA_VISIBLE_DEVICES"] = self.device
         else:
             del os.environ["CUDA_VISIBLE_DEVICES"]  # previously unset
+        self.tempdir_obj.cleanup()
 
     @parameterized.expand([TEST_CASE_1, TEST_CASE_2, TEST_CASE_3])
-    def test_export(self, key_in_ckpt, use_trace):
-        meta_file = os.path.join(os.path.dirname(__file__), "testing_data", "metadata.json")
-        config_file = os.path.join(os.path.dirname(__file__), "testing_data", "inference.json")
-        with tempfile.TemporaryDirectory() as tempdir:
-            def_args = {"meta_file": "will be replaced by `meta_file` arg"}
-            def_args_file = os.path.join(tempdir, "def_args.yaml")
-
-            ckpt_file = os.path.join(tempdir, "model.pt")
-            ts_file = os.path.join(tempdir, "model.ts")
-
-            parser = ConfigParser()
-            parser.export_config_file(config=def_args, filepath=def_args_file)
-            parser.read_config(config_file)
-            net = parser.get_parsed_content("network_def")
-            save_state(src=net if key_in_ckpt == "" else {key_in_ckpt: net}, path=ckpt_file)
-
-            cmd = ["coverage", "run", "-m", "monai.bundle", "ckpt_export", "network_def", "--filepath", ts_file]
-            cmd += ["--meta_file", meta_file, "--config_file", f"['{config_file}','{def_args_file}']", "--ckpt_file"]
-            cmd += [ckpt_file, "--key_in_ckpt", key_in_ckpt, "--args_file", def_args_file]
-            if use_trace == "True":
-                cmd += ["--use_trace", use_trace, "--input_shape", "[1, 1, 96, 96, 96]"]
-            command_line_tests(cmd)
```

Comment:
Part of this test is to check that the command line program also works, this isn't being tested now because you're calling `ckpt_export` directly. This is fine for checking that the function works, but we should have one test at least (in a separate method?) to check that the command line invocation still functions.

Reply:
Humm, I am not sure if I understand correctly, sorry. Would it be necessary to run the full cycle of the test, or would just confirming the command exists and is calling the correct function be enough?

Reply:
What the test originally did was run the command `coverage run -m monai.bundle ckpt_export ...` which I know just calls `ckpt_export`. I think it would be good to still check that the command still works as a separate test. I would add one test method in that runs this command once for the content of one of the tests just to be sure it doesn't get broken.

Reply:
Sure, I understand it was running the command line, but I suppose I don't need to test that the output of `ckpt_export` is the same whether I call `ckpt_export` or use the command line. What I would do is call the command line just to verify it exists and accept the parameters, but I would mock the actual rerun of the `ckpt_export` processes. Would you agree with that approach?

Reply:
Sorry, I don't mean to impose. I was under the impression that this was not less rigorous and would avoid redundant tests. My main concern was the time it took to run all the tests, especially if you don't have a GPU. However, I am only starting to understand the project, so I will follow your perception that that test was correct.

I will revert the changes only to eliminate duplication, but I will keep the use of the CLI. Sorry about the confusion.

Reply:
No worries! We're getting there, the time improvement if worthwhile for sure. Thanks!